File: app/src/main/java/com/rmambulancier/nova/MainActivity.java
package com.rmambulancier.nova;

import android.app.Activity;
import android.content.Intent;
import android.graphics.Color;
import android.graphics.Typeface;
import android.net.Uri;
import android.os.Bundle;
import android.provider.Settings;
import android.view.Gravity;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.ScrollView;
import android.widget.Switch;
import android.widget.TextView;

public final class MainActivity extends Activity {
    private android.content.SharedPreferences prefs;
    private TextView accessText;
    private TextView overlayText;

    @Override public void onCreate(Bundle state) {
        super.onCreate(state);
        prefs = getSharedPreferences("nova", MODE_PRIVATE);
        if (android.os.Build.VERSION.SDK_INT >= 30) getWindow().setDecorFitsSystemWindows(false);
        setContentView(buildUi());
    }

    @Override protected void onResume() {
        super.onResume();
        if (accessText != null) refreshStatus();
    }

    private View buildUi() {
        LinearLayout root = new LinearLayout(this);
        root.setOrientation(LinearLayout.VERTICAL);
        root.setBackgroundColor(Color.rgb(7, 8, 10));
        root.setPadding(dp(22), dp(18), dp(22), dp(24));

        ScrollView scroll = new ScrollView(this);
        scroll.setFillViewport(true);
        LinearLayout content = new LinearLayout(this);
        content.setOrientation(LinearLayout.VERTICAL);
        content.setGravity(Gravity.CENTER_HORIZONTAL);
        scroll.addView(content, new ViewGroup.LayoutParams(-1, -1));

        content.addView(text("NOVA", 34, Color.WHITE, Typeface.DEFAULT_BOLD));
        content.addView(text("Dynamic Spot 4.0", 18, Color.LTGRAY, Typeface.DEFAULT_BOLD), top(2));
        content.addView(text("Android 17 • Pixel • 100 % local", 13, Color.GRAY, Typeface.DEFAULT), top(6));

        LinearLayout status = card();
        accessText = text("Accès notifications : …", 15, Color.WHITE, Typeface.DEFAULT_BOLD);
        overlayText = text("Overlay : …", 15, Color.WHITE, Typeface.DEFAULT_BOLD);
        status.addView(accessText);
        status.addView(overlayText, top(8));
        content.addView(status, top(24));

        content.addView(button("Autoriser l’overlay", () -> {
            startActivity(new Intent(Settings.ACTION_MANAGE_OVERLAY_PERMISSION, Uri.parse("package:" + getPackageName())));
        }), top(14));
        content.addView(button("Autoriser l’accès aux notifications", () -> {
            startActivity(new Intent("android.settings.ACTION_NOTIFICATION_LISTENER_SETTINGS"));
        }), top(10));

        LinearLayout features = card();
        features.addView(text("FONCTIONS", 12, Color.GRAY, Typeface.DEFAULT_BOLD));
        addSwitch(features, "🎵 Média", "Musique, podcasts, pochette, progression", "media", true);
        addSwitch(features, "🔔 Notifications", "Alertes courtes", "notifications", true);
        addSwitch(features, "🔋 Batterie", "Charge et batterie faible", "battery", true);
        content.addView(features, top(22));

        LinearLayout info = card();
        info.addView(text("Architecture V4", 15, Color.WHITE, Typeface.DEFAULT_BOLD));
        info.addView(text("Base Android native uniquement. Aucun Compose, aucune librairie UI, aucun serveur, aucune publicité, aucun analytics.", 13, Color.LTGRAY, Typeface.DEFAULT), top(8));
        content.addView(info, top(14));

        root.addView(scroll, new LinearLayout.LayoutParams(-1, 0, 1f));
        return root;
    }

    private void addSwitch(LinearLayout parent, String title, String subtitle, String key, boolean def) {
        LinearLayout row = new LinearLayout(this);
        row.setOrientation(LinearLayout.HORIZONTAL);
        row.setGravity(Gravity.CENTER_VERTICAL);
        row.setPadding(0, dp(12), 0, dp(4));
        LinearLayout labels = new LinearLayout(this);
        labels.setOrientation(LinearLayout.VERTICAL);
        labels.addView(text(title, 15, Color.WHITE, Typeface.DEFAULT_BOLD));
        labels.addView(text(subtitle, 12, Color.GRAY, Typeface.DEFAULT), top(3));
        row.addView(labels, new LinearLayout.LayoutParams(0, -2, 1f));
        Switch sw = new Switch(this);
        sw.setChecked(prefs.getBoolean(key, def));
        sw.setOnCheckedChangeListener((button, checked) -> prefs.edit().putBoolean(key, checked).apply());
        row.addView(sw);
        parent.addView(row);
    }

    private LinearLayout card() {
        LinearLayout v = new LinearLayout(this);
        v.setOrientation(LinearLayout.VERTICAL);
        v.setPadding(dp(16), dp(15), dp(16), dp(15));
        v.setBackgroundColor(Color.rgb(22, 24, 28));
        return v;
    }

    private TextView button(String label, final Runnable action) {
        TextView v = text(label, 14, Color.BLACK, Typeface.DEFAULT_BOLD);
        v.setGravity(Gravity.CENTER);
        v.setBackgroundColor(Color.WHITE);
        v.setPadding(dp(12), dp(13), dp(12), dp(13));
        v.setOnClickListener(view -> action.run());
        return v;
    }

    private TextView text(String value, float size, int color, Typeface face) {
        TextView v = new TextView(this);
        v.setText(value);
        v.setTextSize(size);
        v.setTextColor(color);
        v.setTypeface(face);
        return v;
    }

    private LinearLayout.LayoutParams top(int px) {
        LinearLayout.LayoutParams p = new LinearLayout.LayoutParams(-1, -2);
        p.topMargin = dp(px);
        return p;
    }

    private int dp(int value) { return Math.round(value * getResources().getDisplayMetrics().density); }

    private void refreshStatus() {
        boolean access = NovaNotificationListenerService.isAccessGranted(this);
        boolean overlay = Settings.canDrawOverlays(this);
        accessText.setText("Accès notifications : " + (access ? "ACTIF" : "À autoriser"));
        overlayText.setText("Overlay : " + (overlay ? "ACTIF" : "À autoriser"));
    }
}

File: app/src/main/java/com/rmambulancier/nova/MainActivity.kt
package com.rmambulancier.nova

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.provider.Settings
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.activity.enableEdgeToEdge
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.Button
import androidx.compose.material3.Card
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Switch
import androidx.compose.material3.Text
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp

class MainActivity : ComponentActivity() {
    private val prefs by lazy { getSharedPreferences("nova", MODE_PRIVATE) }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContent { NovaApp() }
    }

    override fun onResume() {
        super.onResume()
        setContent { NovaApp() }
    }

    private fun notificationAccessEnabled(): Boolean =
        NovaNotificationListenerService.isAccessGranted(this)

    private fun openOverlay() {
        startActivity(Intent(Settings.ACTION_MANAGE_OVERLAY_PERMISSION, Uri.parse("package:$packageName")))
    }

    private fun openNotifications() {
        startActivity(Intent(Settings.ACTION_NOTIFICATION_LISTENER_SETTINGS))
    }

    @androidx.compose.runtime.Composable
    private fun NovaApp() {
        var overlay by remember { mutableStateOf(Settings.canDrawOverlays(this)) }
        var notifications by remember { mutableStateOf(notificationAccessEnabled()) }
        var media by remember { mutableStateOf(prefs.getBoolean("media", true)) }
        var notification by remember { mutableStateOf(prefs.getBoolean("notifications", true)) }
        var battery by remember { mutableStateOf(prefs.getBoolean("battery", true)) }

        Scaffold { padding ->
            Column(
                Modifier.fillMaxSize().verticalScroll(rememberScrollState()).padding(padding).padding(20.dp),
                verticalArrangement = Arrangement.spacedBy(14.dp)
            ) {
                Text("NOVA", style = MaterialTheme.typography.displaySmall)
                Text("Dynamic Spot 2.0 • Android 17", style = MaterialTheme.typography.titleMedium)
                Text("Local, privé, sans publicité, sans compte et sans serveur.")

                Card(Modifier.fillMaxWidth()) {
                    Column(Modifier.padding(18.dp), verticalArrangement = Arrangement.spacedBy(10.dp)) {
                        Text("État", style = MaterialTheme.typography.titleLarge)
                        Text(if (overlay) "✓ Affichage flottant actif" else "○ Affichage flottant à autoriser")
                        Text(if (notifications) "✓ Accès notifications actif" else "○ Accès notifications à autoriser")
                        Row(horizontalArrangement = Arrangement.spacedBy(10.dp)) {
                            Button(onClick = { openOverlay(); overlay = Settings.canDrawOverlays(this@MainActivity) }) { Text("Affichage") }
                            OutlinedButton(onClick = { openNotifications(); notifications = notificationAccessEnabled() }) { Text("Notifications") }
                        }
                    }
                }

                Card(Modifier.fillMaxWidth()) {
                    Column(Modifier.padding(18.dp), verticalArrangement = Arrangement.spacedBy(6.dp)) {
                        Text("Modules", style = MaterialTheme.typography.titleLarge)
                        NovaSwitch("🎵 Musique + contrôles", media) {
                            media = it; prefs.edit().putBoolean("media", it).apply()
                        }
                        NovaSwitch("🔔 Notifications", notification) {
                            notification = it; prefs.edit().putBoolean("notifications", it).apply()
                        }
                        NovaSwitch("🔋 Batterie + charge", battery) {
                            battery = it; prefs.edit().putBoolean("battery", it).apply()
                        }
                    }
                }

                Card(Modifier.fillMaxWidth()) {
                    Column(Modifier.padding(18.dp), verticalArrangement = Arrangement.spacedBy(8.dp)) {
                        Text("Nouveautés V2", style = MaterialTheme.typography.titleLarge)
                        Text("• Dynamic Spot adaptative et plus compacte")
                        Text("• Pochette, titre, artiste et progression média")
                        Text("• Tap = lecture/pause • glissement = précédent/suivant")
                        Text("• Gestion propre des changements de lecteur média")
                        Text("• File d'événements pour éviter le spam des notifications")
                        Text("• Batterie, charge et batterie faible")
                        Text("• Réglages locaux persistants")
                    }
                }

                Spacer(Modifier.height(8.dp))
                Text("Après activation des deux autorisations, lance une musique pour tester NOVA.")
            }
        }
    }

    @androidx.compose.runtime.Composable
    private fun NovaSwitch(label: String, checked: Boolean, onCheckedChange: (Boolean) -> Unit) {
        Row(Modifier.fillMaxWidth(), verticalAlignment = Alignment.CenterVertically) {
            Text(label, Modifier.weight(1f))
            Switch(checked = checked, onCheckedChange = onCheckedChange)
        }
    }
}
